### ISSUE TITLE
broken tbb parallel_for for 64 bit indices

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#7835]
I am getting compile errors like
```
/ssd/deal-git/bundled/tbb-2018_U2/include/tbb/parallel_for.h:116:13: error: no match for call to ‘(const dealii::internal::VectorOperations::TBBReduceFunctor<dealii::internal::VectorOperations::Dot<std::complex<double>, std::complex<double> >, std::complex<double> >) (tbb::blocked_range<unsigned int>&)’
             my_body( r ); 
/ssd/deal-git/include/deal.II/lac/vector_operations_internal.h:1320:7: note:   no known conversion for argument 1 from ‘tbb::blocked_range<unsigned int>’ to ‘const tbb::blocked_range<long long unsigned int>&’
```
I think the blocked_range types need to be ``unsigned int`` even if we have 64 bit indices. I am working on a patch that fixes compilation, but I think we need to check all places carefully.

**Comment by tjhei** [comment]
see #7836, but I think we need to check the types more carefully...

**Comment by tjhei** [comment]
I think this was fixed by #7836. At least I have no recollection of this issue. :shrug:
